src/context-selection/period-selector-bar-item/use-periods.js

```diff
@@ -47,6 +48,43 @@ export default function usePeriods({
             // here, regardless of the period type.
             // + 1 so we include 1970 as well
             yearsCount: yearForGenerating - 1970 + 1,
+        }
+        const periods = generateFixedPeriods(generateFixedPeriodsPayload)
+
+        if (isYearlyPeriodType) {
+            return periods
+        }
+
+        const [lastPeriodOfPrevYear] = generateFixedPeriods({
```

Comment:
I think you can end up in situations where this will be undefined. Like I have openFuturePeriods:30 for a monthly dataset, I select a period in 2025, then I go and select another monthly dataset with no openFuturePeriods. In which case, the app will error out

Reply:
fixed